test/e2e_tests/specs/CriticalFlow/messagesInGroups-TC-8751.spec.ts

```diff
@@ -0,0 +1,194 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {getUser} from 'test/e2e_tests/data/user';
+import {PageManager} from 'test/e2e_tests/pageManager';
+import {getVideoFilePath, getAudioFilePath, getTextFilePath, isAssetDownloaded} from 'test/e2e_tests/utils/asset.util';
+import {getImageFilePath, getLocalQRCodeValue} from 'test/e2e_tests/utils/sendImage.util';
+import {addCreatedUser, removeCreatedUser} from 'test/e2e_tests/utils/tearDown.util';
+import {loginUser} from 'test/e2e_tests/utils/userActions';
+
+import {test, expect} from '../../test.fixtures';
+
+const userA = getUser();
+const userB = getUser();
+const conversationName = 'Critical Group';
+const messageText = 'Hello, this is a test message!';
+const selfDestructMessageText = 'This message will self-destruct in 10 seconds.';
+
+const imageFilePath = getImageFilePath();
+const videoFilePath = getVideoFilePath();
+const audioFilePath = getAudioFilePath();
+const textFilePath = getTextFilePath();
+
+test(
+  'Messages in Groups',
+  {tag: ['@TC-8751', '@crit-flow-web']},
+  async ({pageManager: userAPageManager, api, browser}) => {
+    test.slow(); // Increasing test timeout to 90 seconds to accommodate the full flow
+
+    const {pages: userAPages, modals: userAModals, components: userAComponents} = userAPageManager.webapp;
+
+    const userBContext = await browser.newContext();
+    const userBPage = await userBContext.newPage();
+    const userBPageManager = PageManager.from(userBPage);
+    const {pages: userBPages, modals: userBModals} = userBPageManager.webapp;
+
+    await test.step('Preconditions: Creating preconditions for the test via API', async () => {
+      await api.createPersonalUser(userA);
+      await api.createPersonalUser(userB);
+      addCreatedUser(userA);
+      addCreatedUser(userB);
+      await api.connectUsers(userA, userB);
+    });
+
+    await test.step('Both users log in and open the group', async () => {
+      await Promise.all([
+        (async () => {
+          await userAPageManager.openMainPage();
+          await loginUser(userA, userAPageManager);
+          await userAModals.dataShareConsent().clickDecline();
+          await userAPages.conversationList().clickCreateGroup();
+          await userAPages.groupCreation().setGroupName(conversationName);
+          await userAPages.startUI().selectUsers([userB.username]);
+          await userAPages.groupCreation().clickCreateGroupButton();
+        })(),
+
+        (async () => {
+          await userBPageManager.openMainPage();
+          await loginUser(userB, userBPageManager);
+          await userBModals.dataShareConsent().clickDecline();
+        })(),
+      ]);
```

Comment:
I would suggest to make it more readable.
```suggestion
  const setupUserA = async () => {
    await userAPageManager.openMainPage();
    await loginUser(userA, userAPageManager);
    await userAModals.dataShareConsent().clickDecline();
    await userAPages.conversationList().clickCreateGroup();
    await userAPages.groupCreation().setGroupName(conversationName);
    await userAPages.startUI().selectUsers([userB.username]);
    await userAPages.groupCreation().clickCreateGroupButton();
};

  const setupUserB = async () => {
    await userBPageManager.openMainPage();
    await loginUser(userB, userBPageManager);
    await userBModals.dataShareConsent().clickDecline();
  };
  
  await Promise.all([setupUserA(), setupUserB()]);
```

Reply:
Thanks, will do!